src/toil/lib/conversions.py

```diff
@@ -1,8 +1,8 @@
 """Conversion utilities for mapping memory, disk, core declarations from strings to numbers and vice versa."""
 # TODO: Consolidate all conversion utilities here to use the same functions:
 #  src/toil/lib/humanize.py (bytes2human; human2bytes)
```

Comment:
@adamnovak @DailyDreaming I wasn't sure how to consolidate [`src/toil/lib/humanize.py `](https://github.com/DataBiosphere/toil/blob/master/src/toil/lib/humanize.py) here since the entire file was taken elsewhere under a different license.

Reply:
@w-gao `src/toil/lib/humanize.py` can be either deleted/replaced, or become the one true source of conversions.  The same goes for `bytes_in_unit()` and `convert_units()`.  The idea is to simplify the code to make it dirt simple.  We have three ways to turn `10MB` into `10000000` bytes, and we only need one.  The issue is one of design, and it involves choosing one of the three methods we currently use to convert units and bytes: `MemoryString`, `humanize.py`, or the `bytes_in_unit()`/`convert_units()` functions, and deleting the other two, then replacing everything with one way of doing things.

Reply:
Or coming up with a fourth way you think is better and deleting the all of the current three methods.  It's your choice.

Reply:
While I'm not recommending using [src/toil/lib/humanize.py](https://github.com/DataBiosphere/toil/blob/master/src/toil/lib/humanize.py) (variable names like `iec_ext` are not immediately clear) it's under an MIT license, so feel free to modify it (though if you do and we only use this method, be sure to add your name and that you've modified it under the comment with `Giampaolo Rodola`'s name).

Reply:
@DailyDreaming I've rewritten `toil.lib.humanize`'s `human2bytes()` and `bytes2human()` functions to use `bytes_in_unit() / convert_units()`, and moved them to `toil.lib.conversions`. I feel like these functions are still useful to keep, since `bytes_in_unit() / convert_units()` are too basic to be useful.

As for `MemoryString`, I made it to use the `human2bytes()` function instead of using its own logic, but I didn't feel comfortable removing it entirely because I couldn't find where it is used in Toil other than [here](https://github.com/DataBiosphere/toil/blob/master/src/toil/batchSystems/abstractGridEngineBatchSystem.py#L301), but I wasn't sure where `self.maxMEM` is used...

Reply:
@w-gao  Sadly, I think that's really the only place it's used and so we should probably remove it.